modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/connectors/data/sql/client/AbstractSQLAction.java

```diff
@@ -57,8 +57,14 @@ protected void executeQuery(Context context, SQLConnector connector, String quer
         PreparedStatement stmt = null;
         ResultSet rs = null;
         try {
+            boolean mysql = connector.getDatabaseName().contains("mysql");
             conn = connector.getSQLConnection();
-            stmt = conn.prepareStatement(query);
+            if (mysql) {
```

Comment:
Shall we refactor and extract this logic out to another method, to create the PreparedStatement. So we can update this logic later if needed, without mixing it with the rest of the code here.